### PR TITLE
export C interface, started mobile app

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: rfcs/2022-01-26-mobile-app.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  - [types from Agent/Protocol.hs](https://github.com/simplex-chat/simplexmq/blob/master/src/Simplex/Messaging/Agent/Protocol.hs)).
+  
+This package has its [own sqlite database file](https://github.com/simplex-chat/simplexmq/tree/master/migrations) - as v1 was not backwards compatible migrations are restarted - where it stores all encryption and signing keys, shared secrets, servers and queue addresses - effectively it completely abstracts the network away from chat application, providing an API to manage logical duplex connections.
+
+Simplex-chat library is what we will use from the app:
+  - command type [ChatCommand in Chat.hs](https://github.com/simplex-chat/simplex-chat/blob/master/src/Simplex/Chat.hs#L72) that UI can send to it
+  - UI sends these commands via TBQueue that `inputSubscriber` reads in forever look and sends to `processChatCommand`. There is a hack that `inputSubscriber` not only reads commands but also shows them in the view, depending on the commands.
```
Comment (efim-poberezkin) [inline rfcs/2022-01-26-mobile-app.md:43]:
```suggestion
  - UI sends these commands via TBQueue that `inputSubscriber` reads in forever loop and sends to `processChatCommand`. There is a hack that `inputSubscriber` not only reads commands but also shows them in the view, depending on the commands.
```

File: rfcs/2022-01-26-mobile-app.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  - collection of [view functions in Chat/View.hs](https://github.com/simplex-chat/simplex-chat/blob/master/src/Simplex/Chat/View.hs) to reflect all events in view.
+
+This package also creates its own [database file](https://github.com/simplex-chat/simplex-chat/tree/master/migrations) where it stores references to agent connections managed by the agent, and how they may to contacts, groups, and file transmissions.
```
Comment (efim-poberezkin) [inline rfcs/2022-01-26-mobile-app.md:46]:
```suggestion
This package also creates its own [database file](https://github.com/simplex-chat/simplex-chat/tree/master/migrations) where it stores references to agent connections managed by the agent, and how they map to contacts, groups, and file transmissions.
```

File: rfcs/2022-01-26-mobile-app.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+Overall, the second option seems a bit simpler and cleaner, if we agree to go this route we will refactor `processChatCommand` to expose its parts that process different commands as independent functions.
+
+On another hand, it might be easier to grow chat API if this is passed via a single function and serialized as strings (e.g. as JSON, to have it more universal) - it would also might give us an API for a possible future chat server that works with thin, UI-only clients.
+
+In both cases, we should split `processChatCommand` (or the functions it calls) into a separate module, so it does not have code that is not used from the app.
+
+**Proposal**
+
+Use option 2 to send commands from UI to chat, encoding/decoding commands as strings with a tag in the beginning (TBC binary, text or JSON based - encoding will have to be replicated in UI land; both encoding and decoding is needed in Haskell land to refactor terminal chat to use this layer as well, so we have a standard API for all implementations).
```
Comment (efim-poberezkin) [inline rfcs/2022-01-26-mobile-app.md:64]: isn't it the option 1 that is proposed?

File: src/Simplex/Chat.hs [hunk 4]
```diff
@@ -127,40 +126,50 @@ logCfg :: LogConfig
...
+runSimplexChat ct = runReaderT $ do
+  Just user <- readTVarIO =<< asks currentUser
+  whenM (asks firstTime) . liftIO . printToTerminal ct $ chatWelcome user
+  raceAny_ [runTerminalInput ct, runTerminalOutput ct, runChatController]
 
 runChatController :: (MonadUnliftIO m, MonadReader ChatController m, MonadFail m) => m ()
-runChatController =
+runChatController = do
+  q <- asks outputQ
+  let toView = atomically . writeTBQueue q
```
Comment (epoberezkin) [inline src/Simplex/Chat.hs:169]: This function can be refactored out, this value is the only one used

**Pull request merged**